### DISCUSSION TITLE
Does fetchSockets return all sockets from all nodes?

### DISCUSSION THREAD

**Discussion by nicholaswmin** [socketio/socket.io#4513] [Q&A] [answered]
We're using [@socket.io/redis-adapter][redis-adapter] because we're running socket.io on **4 dynos/nodes** and we want this to work in a horizontally scalable way.

We need to fetch all the sockets in a particular room *regardless* of the dyno/node they connected to.

The question:

Does `const sockets = await io.of('/').in('room1').fetchSockets()` fetch all the sockets **across all nodes/dynos** for a room? Our experiments indicate that this is indeed the case but we'd like to confirm it :)

[redis-adapter]: https://socket.io/docs/v4/redis-adapter/

**Comment by darrachequesne** [comment, marked as answer]
@nicholaswmin yes it does! You can also use the `local` flag if you need it:

```js
// all sockets across all nodes
const allSockets = await io.in("room1").fetchSockets();

// all sockets on the node
const sockets = await io.local.in("room1").fetchSockets();
```

Reference: https://socket.io/docs/v4/server-api/#serverfetchsockets
